sympy/polys/polytools.py

```diff
@@ -6218,8 +6216,44 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['polys'])
+    opt = options.build_options(gens, args)
+    e = together(f)
+    gen_set = set()
+    coeff = 1
+    facs = []
+    for fac in Mul.make_args(e):
+        b, e = (fac.base, fac.exp) if fac.is_Pow else (fac, S.One)
+        _opt = opt.clone(dict(expand=True))
+        if e < 0:
+            raise PolynomialError("a univariate polynomial expected")
+        try:
+            poly = _poly_from_expr(b, _opt)
```

Comment:
Could this be `poly, _ = ...`? The generators are also an attribute of `poly`: `poly.gens`.